### PR TITLE
CI build checks for error and warning in apache logs

### PR DESCRIPTION
**Pull request by jpwhite4**
Same as https://github.com/ubccr/xdmod/pull/1726 but for the supremm module build

This pull request will need to be merged after the changes in  https://github.com/ubccr/xdmod/pull/1728 are merged (because these changes depend on a fix that is  in https://github.com/ubccr/xdmod/pull/1728)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -135,13 +138,18 @@ jobs:
             - run:
                 name: Run UI Tests
                 command: "PATH=/usr/local/node-v10.24.1-linux-x64/bin:$PATH $XDMOD_SRC_DIR/tests/ui/runtests.sh --headless --log-junit ~/phpunit"
+            - run:
+                name: Ensure that no unexpected Apache errors were generated
+                command: test ! -e /var/log/php-fpm/www-error.log
```
Comment (aaronweeden) [inline .circleci/config.yml:143]: Might be helpful to print the errors if there are any:
```suggestion
                command: >
                    if [ -e /var/log/php-fpm/www-error.log ]; then
                        cat /var/log/php-fpm/www-error.log;
                        false;
                    fi
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -121,6 +121,9 @@ jobs:
             - run:
                 name: Run UI Tests
                 command: scl enable rh-nodejs6 "$XDMOD_SRC_DIR/tests/ui/runtests.sh --headless --log-junit ~/phpunit"
+            - run:
+                name: Ensure that no unexpected Apache errors were generated
+                command: test `fgrep -v ' [ssl:warn] ' /var/log/xdmod/apache-error.log | wc -l` = 0
```
Comment (aaronweeden) [inline .circleci/config.yml:126]: Might be helpful to print the errors if there are any:
```suggestion
                command: >
                    errors=`fgrep -v ' [ssl:warn] ' /var/log/xdmod/apache-error.log`;
                    if [ -n "${errors}" ];
                    then
                        echo "${errors}";
                        false;
                    fi
```

Comment (jpwhite4) [reply .circleci/config.yml:126]: This file is copied to the build artifacts (and hence the contents appear in the circle ci interface) so no need to redundantly output the errors twice.

File: .circleci/config.yml (hunk 1 above)
Comment (jpwhite4) [reply .circleci/config.yml:143]: This file is copied to the build artifacts (and hence the contents appear in the circle ci interface) so no need to redundantly output the errors twice.

File: tests/component_tests/bootstrap.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                    'lib',
+                    str_replace('\\', '/', $className) . '.php'
+                )
+            );
+        } else {
+            $classPath
+                = $dir
+                . '/lib/'
+                . str_replace('\\', '/', $className)
+                . '.php';
```
Comment (aaronweeden) [inline tests/component_tests/bootstrap.php:35]: Duplicate code of lines 26–27; could be moved after this if/else.

File: .circleci/config.yml [hunk 4]
```diff
@@ -97,7 +97,9 @@ jobs:
           command: ./tests/integration_tests/scripts/validate.sh
       - run:
           name: Make sure that the Composer Test Dependencies are installed
-          command: composer install -d $XDMOD_SRC_DIR --no-progress
+          command: |
+              composer install -d $XDMOD_SRC_DIR --no-progress
+              composer install --no-progress
```
Comment (aaronweeden) [inline .circleci/config.yml:102]: Is this command necessary if already done on line 69?

**Review by aaronweeden — APPROVED** [review]

**Pull request merged**